cognite/client/_api/datapoints.py

```diff
@@ -1520,7 +1520,7 @@ def _validate_and_format_datapoints(
 
     def _bin_datapoints(self, dps_object_list: list[dict[str, Any]]) -> list[list[dict[str, Any]]]:
         for dps_object in dps_object_list:
-            for i in range(0, len(dps_object["datapoints"]), self.limit):
+            for i in range(len(dps_object["datapoints"]), self.limit):
```

Comment:
This does not work, you are now trying to loop from `len(...)` to `self.limit` instead of jumping `self.limit` at the time.

Reply:
Whoops! Thanks for the catch, I forgot that you can set negative indices and that's why it won't run through any of it. Like below.

```
>>> for x in range(89, 2):
...     print(x)
...
>>> for x in range(89, 2, -1):
...     print(x)
...
89
88
87
86
85
...
```